File: solutions/go/grains/1/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (count uint64, err error) {
	if number < 1 || number > 64 {
		err = errors.New("number can't be less than 1 or more than 64")
	} else {
		y := float64(number - 1)
		count = uint64(math.Pow(2, y))
	}
	return
}

func Total() uint64 {
	return math.MaxUint64
}

File: solutions/typescript/grains/1/grains.ts
export const square = (nbr: number): bigint => {
  if (nbr < 1 || nbr > 64) {
    throw new Error('The requested square must be between 1 and 64 (inclusive)')
  }

  return BigInt(2 ** (nbr - 1))
}

export const total = (): bigint => BigInt(2 ** 64) - 1n
